auth/auth/auth.py

```diff
@@ -414,23 +579,29 @@ async def delete_user(request, userdata):  # pylint: disable=unused-argument
     id = post['id']
     username = post['username']
 
-    n_rows = await db.execute_update(
-        '''
-UPDATE users
-SET state = 'deleting'
-WHERE id = %s AND username = %s;
-''',
-        (id, username),
-    )
-    if n_rows != 1:
-        assert n_rows == 0
-        set_message(session, f'Delete failed, no such user {id} {username}.', 'error')
-    else:
+    try:
+        await _delete_user(db, username, id)
         set_message(session, f'Deleted user {id} {username}.', 'info')
+    except UnknownUser:
+        set_message(session, f'Delete failed, no such user {id} {username}.', 'error')
 
     return web.HTTPFound(deploy_config.external_url('auth', '/users'))
 
 
+@routes.delete('/api/v1alpha/users/{user}')
+@rest_authenticated_developers_only
+async def rest_delete_user(request: web.Request, userdata):  # pylint: disable=unused-argument
+    db = request.app['db']
+    username = request.match_info['user']
+
+    try:
+        await _delete_user(db, username, None)
+    except UnknownUser as e:
+        return e.http_response()
```

Comment:
I think aiohttp warns if you return an error instead of `raise`-ing it.

Reply:
From the docs:
https://docs.aiohttp.org/en/v0.14.3/web.html#exceptions

You can do either. I'll fix these to raise, but we ran into problems with the session message showing up if we raised errors.